quandl/model/model_base.py

```diff
@@ -1,21 +1,26 @@
 class ModelBase(object):
-    def __init__(self, raw_data, **options):
-        self.raw_data = raw_data
+    def __init__(self, code, raw_data=None, **options):
+        self.code = code
+        self._raw_data = raw_data
+        self.options = options
 
     def to_list(self):
-        if isinstance(self.raw_data, dict):
-            return list(self.raw_data.values())
-        return self.raw_data
+        if isinstance(self.__get_raw_data__(), dict):
+            return list(self.__get_raw_data__().values())
+        return self.__get_raw_data__()
 
     def __getattr__(self, k):
         if k[0] == '_':
             raise AttributeError(k)
-        elif k in self.raw_data:
-            return self.raw_data[k]
+        elif k in self.__get_raw_data__():
+            return self.__get_raw_data__()[k]
         raise AttributeError(k)
 
     def __getitem__(self, k):
-        return self.raw_data[k]
+        return self.__get_raw_data__()[k]
 
     def data_fields(self):
-        return list(self.raw_data.keys())
+        return list(self.__get_raw_data__().keys())
+
+    def __get_raw_data__(self):
```

Comment:
Edit: I see how this might be done on purpose - if GetOperation is inherited from, then we would have the http request GET operation, otherwise we would just simply return `_raw_data`?